### PR TITLE
Setup publishing for autofill-parser

### PR DESCRIPTION
**Pull request by msfjarvis**
## :loudspeaker: Type of change

- [ ] Bugfix
- [x] New feature
- [x] Enhancement
- [ ] Refactoring
- [ ] Dependency updates

## :scroll: Description
Since every public Maven repository proved to be hellspawn, and GitHub Package Registry requires
authentication for everything, including pulling public artifacts, I decided to host my own.

TODO:

- [ ] ~~Set up dokka to generate HTML docs~~
- [ ] ~~Publish generated docs~~
- [ ] Include sources JAR in the published artifacts

## :bulb: Motivation and Context
I want to make autofill-parser v1.0.0 and APS v1.12.0 available at the same time so getting it
onto a publicly accessible Maven repository is a necessity.

## :green_heart: How did you test it?
Verified I can resolve the dependency when using the configuration described in the project README.

## :pencil: Checklist

- [x] I formatted the code with the IDE's reformat action (Ctrl + Shift + L/Cmd + Shift + L)
- [x] I reviewed submitted code
- [ ] I added a [CHANGELOG](CHANGELOG.md) entry if applicable

### CODE REVIEW

**Comment by msfjarvis** [comment]
@FabianHenneke do you wanna do a documentation update pass before v1.12.0 is due or should we defer publishing of docs to sometime later?

**Comment by fmeum** [comment]
> @FabianHenneke do you wanna do a documentation update pass before v1.12.0 is due or should we defer publishing of docs to sometime later?

I won't be able to do this until Thursday, but it's on my list.

**Comment by msfjarvis** [comment]
> > @FabianHenneke do you wanna do a documentation update pass before v1.12.0 is due or should we defer publishing of docs to sometime later?
> 
> I won't be able to do this until Thursday, but it's on my list. 

Alright, I'll skip docs then.

**Comment by msfjarvis** [comment]
I'm keeping publishing manual for now, but will soon automate it with a `workflow_dispatch` trigger so new updates can be pushed by maintainers without my involvement.

Edit: don't squash this, I want to preserve the history here.

File: gradle.properties [hunk 1]
```diff
@@ -2,24 +2,50 @@
 # Copyright © 2014-2020 The Android Password Store Authors. All Rights Reserved.
 # SPDX-License-Identifier: GPL-3.0-only
 #
+org.gradle.parallel=true
+org.gradle.configureondemand=true
+org.gradle.caching=true
```
Comment (msfjarvis) [inline gradle.properties:7]: Standard stuff

File: gradle.properties [hunk 2]
```diff
@@ -2,24 +2,50 @@
...
 
-# AndroidX
-android.enableJetifier=false
-android.useAndroidX=true
+# Enabling filesystem watching
+org.gradle.vfs.watch=true
+org.gradle.unsafe.watch-fs=true
```
Comment (msfjarvis) [inline gradle.properties:11]: First one enables it for Gradle 6.7, second one for Gradle 6.6. We'll eventually drop the 6.6 variant, but I couldn't risk F-Droid being pissy about early bird Gradle.

File: gradle.properties [hunk 3]
```diff
@@ -2,24 +2,50 @@
...
-# https://jakewharton.com/increased-accuracy-of-aapt2-keep-rules/
-android.useMinimalKeepRules=true
+# Kotlin code style
+kotlin.code.style=official
+
+# Enable incremental annotation processor for KAPT
+kapt.incremental.apt=true
+
+# Turn off AP discovery in compile path to enable compile avoidance
+kapt.include.compile.classpath=false
```
Comment (msfjarvis) [inline gradle.properties:26]: Not currently using `kapt` but there might be some possibility in the future, so better safe than sorry.

File: gradle.properties [hunk 4]
```diff
@@ -2,24 +2,50 @@
...
+
+# Enable AndroidX
+android.useAndroidX=true
 
-# Enable rudimentary R class namespacing where each library only contains
+# Enable non-transitive R class namespacing where each library only contains
 # references to the resources it declares instead of declarations plus all
 # transitive dependency references.
-android.namespacedRClass=true
+android.nonTransitiveRClass=true
```
Comment (msfjarvis) [inline gradle.properties:37]: Same feature, just improved and renamed.

File: gradle.properties [hunk 5]
```diff
@@ -2,24 +2,50 @@
...
 
-# Gradle FS watching
-org.gradle.unsafe.watch-fs=true
+# Only keep the single relevant constructor for types mentioned in XML files
+# instead of using a parameter wildcard which keeps them all.
+android.useMinimalKeepRules=true
+
+# Enable resource optimizations for release build
+android.enableResourceOptimizations=true
```
Comment (msfjarvis) [inline gradle.properties:44]: Unused before AGP 4.2.0, adding for future-proofing.

File: gradle.properties [hunk 6]
```diff
@@ -2,24 +2,50 @@
...
+
+# Default Android build features
+android.defaults.buildfeatures.buildconfig=false
+android.defaults.buildfeatures.aidl=false
+android.defaults.buildfeatures.renderscript=false
+android.defaults.buildfeatures.resvalues=false
+android.defaults.buildfeatures.shaders=false
```
Comment (msfjarvis) [inline gradle.properties:51]: Setting logical defaults which would get unwieldy to configure on a per-project basis once we started modularizing more aggressively.

**Review by fmeum — APPROVED** [review]
Great stuff, thanks for taking it up!

I will let you merge this.

**Pull request merged**